src/librustc/ty/util.rs

```diff
@@ -453,17 +453,6 @@ impl<'a, 'gcx, 'tcx> TypeVisitor<'tcx> for TypeIdHasher<'a, 'gcx, 'tcx> {
                 // Hash region and builtin bounds.
                 data.region_bound.visit_with(self);
```

Comment:
This and `data.principal.visit_with(self);` above it are unnecessary because automatic recursion will hit them anyway.